### PR TITLE
Updated translation function

### PR DESCRIPTION
**Pull request by sandeep-deriv**

### CODE REVIEW

File: src/common/i18n.js [hunk 1]
```diff
@@ -44,9 +44,11 @@ export const translate = (input, params = []) => {
         const stringToBeTranslated = input.replace(/\{\$({0-9])\}/gi, '%$1');
         let translatedString = i18nTranslate(stringToBeTranslated);
         params.forEach((replacement, index) => {
-            translatedString = translatedString.replaceAll(`\{\$${index}\}`, replacement);
+            if (translatedString && typeof translatedString === 'string') {
+                translatedString = translatedString.replaceAll(`\{\$${index}\}`, replacement);
+            }
         });
-        return RenderHTML(translatedString);
+        return RenderHTML(translatedString || '');
```
Comment (salarhfz-fs) [inline src/common/i18n.js:51]: Sounds like having both `translatedString &&` and `translatedString || ''` is redundant here.
Am I correct? Could we remove one and keep just the other?

File: src/common/i18n.js [hunk 2]
```diff
@@ -44,9 +44,11 @@ export const translate = (input, params = []) => {
         const stringToBeTranslated = input.replace(/\{\$({0-9])\}/gi, '%$1');
         let translatedString = i18nTranslate(stringToBeTranslated);
         params.forEach((replacement, index) => {
-            translatedString = translatedString.replaceAll(`\{\$${index}\}`, replacement);
+            if (translatedString && typeof translatedString === 'string') {
+                translatedString = translatedString.replaceAll(`\{\$${index}\}`, replacement);
```
Comment (mehrzad-fs) [inline src/common/i18n.js:48]: is there any situation that the `translatedString` is not `string`
if so, do we need to handle this situation?

**Review by mehrzad-fs — COMMENTED** [review]
this is out of your updates scope but it's a suggestion to make code cleaner 
we have `params.length` condition. we can check `!params.length` then check the rest of codes 
```
export const translate = (input, params = []) => {
    if (!params.length) {
        return i18nTranslate(input);
    }
        const stringToBeTranslated = input.replace(/\{\$({0-9])\}/gi, '%$1');
        let translatedString = i18nTranslate(stringToBeTranslated);
        params.forEach((replacement, index) => {
            if (translatedString && typeof translatedString === 'string') {
                translatedString = translatedString.replaceAll(`\{\$${index}\}`, replacement);
            }
        });
        return RenderHTML(translatedString || '');
    }
};
```

File: src/common/i18n.js (hunk 2 above)
Comment (sandeep-deriv) [reply src/common/i18n.js:48]: Yes, I think sometime somehow the translateString in not string and something else.
Its been caught in the trackjs that r.replaceAll is not a function.
[trackjs](https://my.trackjs.com/errors/filtered?hash=d79fa8f299d70d61)
[image](https://user-images.githubusercontent.com/90243468/148722496-98fa0c29-915c-427e-92f1-017722c35966.png)


So if translateString is null or undefined then it should result `cannot read property replaceAll of null or undefined`.
instead its giving `r.replaceAll is not a function`.

I was unable to reproduce this trackjs error but I thought adding a type check before replaceAll would be a good idea.

Let me know your opinion about this.

File: src/common/i18n.js (hunk 1 above)
Comment (sandeep-deriv) [reply src/common/i18n.js]: Removed.

**Review by mehrzad-fs — APPROVED** [review]

**Pull request merged**